_posts/camazotz/2016-04-29-NavaneetsFinalProject.md

```diff
@@ -0,0 +1,67 @@
+
+When printing out the bar graph of asterisks for the number of votes each candidate has, I didn't want to print out the actual number of votes (since that would mean hundreds of thousands of asterisks). So I decided to scale all the votes between 0 and 10. After doing the multiplication and division necessary for implementing this scaling factor, I had to round the number to an integer. I found the method to do that [here](http://stackoverflow.com/questions/31818050/python-2-7-round-number-to-nearest-integer).
+
+When printing out the asterisks in a loop, I noticed that the print function was automatically inserting a newline at the end. I found a way to disable that by putting end='' on stackoverflow [here](http://stackoverflow.com/questions/12032214/python-3-print-new-output-on-same-line).
+
+In order to get the asterisks to line up when printing in the loop, I used format(string, '<30') to align the string to the left and extend space out for 30 characters. I found that method [here](https://www.safaribooksonline.com/library/view/python-cookbook-3rd/9781449357337/ch02s13.html).
+
+The input does not take keyword arguments error when trying to have variable name in input call so I added ‘+’ to add string instead of comma and got rid of the ‘sep’ parameter.
+
+To exit the program, I use sys.exit() after importing sys module.
+
+To create fake data, I had to find a way to randomize excel data so I used the method outlined [here](http://answers.microsoft.com/en-us/office/forum/office_2007-excel/how-to-randomize-data-in-excelspreadsheets/399ec67b-a443-e011-9bac-78e7d160ad4e?auth=1).
+
+Retrospective/Looking ahead:
+
+Next time I work on something like this I think I would think a bit more critically about the requirements for the functionality before beginning to code. This would have saved me the time that I spent on redesigning the dictionaries. I would have also begun modularizing the design from the beginning because waiting until the end to modularize would require some tweaking of all of the methods (which assume global variables in a non-modularized design), which would be a pain as the project scales up.
+
+Unfortunately I didn't get a chance to look into using Cloud 9 (I used PyCharm instead) because I got overwhelmed by other project deadlines this week but I think it would be helpful to get comfortable using it so I'll look into it over summer. Now that I'm pretty comfortable using dictionaries and reading from datasets, I think I might tackle some of the challenges on Kaggle and start building my data analysis skills.
+
+As I stated in my project proposal, I feel that a lot of the functions that I had to enable using dictionaries could have probably been acheived a lot more efficiently if I had used a database, so that is the next thing I will look into. My plan is to look into using the MongoDB Python user interface and perhaps trying to replicate the same functionality in this app using the dictionary.
```

Comment:
Good observation.  Dictionaries are like little lightweight databases that don't persist from run to run.  There is an example in our text that we didn't get to that uses sqlite to store things- check it out.